### PR TITLE
feat: [CHK-2982] Added npg client tracing

### PR DESCRIPTION
**Pull request by petretiandrea**
This PR adds npg client tracing capability, like eCommerce do.

#### List of Changes
- added span for each npg client api call with custom span name
- added npg error code to spans
- added correlation id to spans

#### How Has This Been Tested?
In DEV by onboarding a new Wallet
Screenshot 2024-06-26 alle 11 42 21
Screenshot 2024-06-26 alle 11 42 39

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/client/NpgClient.kt [hunk 1]
```diff
@@ -36,74 +53,128 @@ class NpgClient(
...
+        ) { span, operation ->
+            logger.info("getCardData with correlationId: $correlationId")
+            Try.of {
+                    npgWebClient.pspApiV1BuildCardDataGet(
+                        correlationId,
+                        npgPaypalPspApiKeysConfig.defaultApiKey,
+                        sessionId
+                    )
+                }
+                .getOrElseGet { Mono.error(it) }
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/wallet/client/NpgClient.kt:99]: Is `Try` necessary? Exceptions should be already caught in the reactive chain

Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/wallet/client/NpgClient.kt:99]: In other words, these should be equivalent, right?
```kotlin
fun foo(): Mono<Integer> {
    Mono.just(0).map {
        throw RuntimeException()
    }
}

val m = Try.of { foo() }.getOrElseGet { Mono.error(it) }
val m2 = foo()
```

File: src/test/kotlin/it/pagopa/wallet/client/NpgClientTest.kt [hunk 2]
```diff
@@ -172,7 +192,9 @@ class NpgClientTest {
                     500,
                     "statusText",
                     HttpHeaders.EMPTY,
-                    ByteArray(0),
+                    objectMapper.writeValueAsBytes(
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/wallet/client/NpgClientTest.kt:195]: add verification against `spanBuilder.setAttribute` method being invoked with expected NPG response and HTTP error codes for this and below test

**Review by pietro-tota — DISMISSED** [review]
Well done, just a trivial issue 🚀

**Review by giovanniberti — APPROVED** [review]
🚀

File: src/main/kotlin/it/pagopa/wallet/client/NpgClient.kt (hunk 1 above)
Comment (petretiandrea) [reply src/main/kotlin/it/pagopa/wallet/client/NpgClient.kt]: Solved by https://github.com/pagopa/pagopa-wallet-service/pull/167/commits/3a598cdb763f8f2a6bfe1ecebaede63c01311e9d

**Review by giovanniberti — APPROVED** [review]

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**